src/Promitor.Tests.Unit/Agents/ResourceDiscovery/AzureResourceGroupsDiscoveryBackgroundJobTests.cs

```diff
@@ -0,0 +1,79 @@
+using Promitor.Agents.ResourceDiscovery.Graph.Repositories.Interfaces;
+using Promitor.Agents.ResourceDiscovery.Scheduling;
+using Promitor.Core.Metrics.Interfaces;
+using System.Collections.Generic;
+using System.Threading;
+using System.Threading.Tasks;
+using Xunit;
+
+namespace Promitor.Tests.Unit.Agents.ResourceDiscovery
+{
+    public class AzureResourceGroupsDiscoveryBackgroundJobTests
+    {
+        private readonly Mock<IAzureResourceRepository> _azureResourceRepository;
+        private readonly Mock<ISystemMetricsPublisher> _systemMetricsPublisher;
+        private readonly AzureSubscriptionDiscoveryBackgroundJob _azureResourceGroupsDiscoveryBackgroundJob;
+        private readonly Promitor.Core.Contracts.PagedPayload<AzureSubscriptionInformation> firstPage;
+        private readonly Promitor.Core.Contracts.PagedPayload<AzureSubscriptionInformation> secondPage;
+
+        public AzureResourceGroupsDiscoveryBackgroundJobTests()
+        {
+            _azureResourceRepository = new Mock<IAzureResourceRepository>(MockBehavior.Strict);
+            _systemMetricsPublisher = new Mock<ISystemMetricsPublisher>(MockBehavior.Strict);
+            var logger = new Mock<ILogger<AzureSubscriptionDiscoveryBackgroundJob>>(MockBehavior.Loose);
+            _azureResourceGroupsDiscoveryBackgroundJob = new AzureSubscriptionDiscoveryBackgroundJob("jobName", _azureResourceRepository.Object, _systemMetricsPublisher.Object, logger.Object);
+
+            firstPage = new Promitor.Core.Contracts.PagedPayload<AzureSubscriptionInformation>()
+            {
+                PageInformation = new Promitor.Core.Contracts.PageInformation { CurrentPage = 1, PageSize = 1, TotalRecords = 1 },
+                Result = new List<AzureSubscriptionInformation>() { new AzureSubscriptionInformation { TenantId = "TenantId", Id = "ID", Name = "Name" } }
+            };
+            secondPage = new Promitor.Core.Contracts.PagedPayload<AzureSubscriptionInformation>()
+            {
+                PageInformation = new Promitor.Core.Contracts.PageInformation { CurrentPage = 2, PageSize = 1, TotalRecords = 1 },
+                Result = new List<AzureSubscriptionInformation>() { new AzureSubscriptionInformation { TenantId = "TenantId", Id = "ID", Name = "Name" } }
+            };
+            _azureResourceRepository.Setup(r => r.DiscoverAzureSubscriptionsAsync(1000, 1)).ReturnsAsync(firstPage);
+            _azureResourceRepository.Setup(r => r.DiscoverAzureSubscriptionsAsync(1000, 2)).ReturnsAsync(secondPage);
+            _systemMetricsPublisher.Setup(p => p.WriteGaugeMeasurementAsync(It.IsAny<string>(), It.IsAny<string>(), 1, It.IsAny<Dictionary<string, string>>(), true)).Returns(Task.CompletedTask);
+        }
+
+        [Fact]
+        public async Task AzureResourceGroupsDiscoveryBackgroundJob_Execution_Succeeds()
+        {
+            // Act
+            await _azureResourceGroupsDiscoveryBackgroundJob.ExecuteAsync(CancellationToken.None);
```

Comment:
We might want to add asserts on the output here, thoughts?

Reply:
Since it's an empty Task I don't see what asserts would be useful, any idea?
The logic of this method is tested by the verify bellow

Reply:
Nah good point, thanks!